### DISCUSSION TITLE
Blending in v9

### DISCUSSION THREAD

**Discussion by aslmbk** [visgl/deck.gl#9024] [General]
Hi all! I wanted to set up AdditiveBlending on ArcLayer.

In the previous version everything worked great, but after I updated to the latest version, blending stopped working

Here is the code

```js
    new ArcLayer<ArcLike>({
      id: 'arcs-layer',
      data,
      getSourcePosition: (a) => a.start,
      getTargetPosition: (a) => a.end,
      getSourceColor: (a) => getColor(a.type),
      getTargetColor: (a) => getColor(a.type),
      getWidth: 1,
      parameters: {
        depthWriteEnabled: false,
        blendColorSrcFactor: 'src-alpha',
        blendAlphaSrcFactor: 'src-alpha',
        blendColorDstFactor: 'one',
        blendAlphaDstFactor: 'one',
        blendColorOperation: 'add',
        blendAlphaOperation: 'add',
        // depthTest: false,
        // blendFunc: [GL.SRC_ALPHA, GL.ONE],
        // blendEquation: GL.FUNC_ADD,
      },
    }),
```

**Comment by ibgreen** [comment]
There is a known issue in luma v9 with blending not being enabled in WebGL by the blending props. Maybe you can do it manually `device.gl.enable(device.gl.BLEND)` if you can find the right place to inject it. Maybe `layer.props.onBeforeRender` or something like that?

**Comment by Pessimistress** [comment]
Seems working for me: https://codepen.io/Pessimistress/pen/XWLmRxZ?editors=0010
